### ISSUE TITLE
can't recover or destroy job in `ProvisioningAndChecking`  state 

### ISSUE THREAD

**Issue by wangwang** [oam-dev/terraform-controller#169]
1.  create `configurations.terraform.core.oam.dev` using bad  writeConnectionSecretToRef.name format, e.g: `foo-bar_abc_def`

2.  terraform apply job will successfully create cloud resources,  and then fail to create conn-secret since we use bad secret name

```
Secret \"foo-bar_abc_def\" is invalid: metadata.name: Invalid value: \"foo-bar_abc_def\": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphan...
```

3. delete this app, but destroy-job will never be created:

```
"error": "continue reconciling to destroy cloud resource: Destroy could not complete and needs to wait for Provision to complete first: Cloud resources are being provisioned and provisioning status is checking...",
 "errorVerbose": "Destroy could not complete and needs to wait for Provision to complete first:
Cloud resources are being provisioned and provisioning status is checking...
```

**Comment by zzxwill** [comment]
@wangwang Much appreciated for reporting this issue. This issue should be reported as a separate state. When the resource is created and creating secret failed, the resource can allow to be deleted.

**Comment by wangwang** [comment]
fixed in #181
